pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -18,25 +18,68 @@ package machinesync
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
+	configv1 "github.com/openshift/api/config/v1"
 	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	"github.com/openshift/cluster-capi-operator/pkg/util"
 	"k8s.io/apimachinery/pkg/runtime"
+	awscapimachinev1beta1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta1"
+	capimachinev1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	capiNamespace string = "openshift-cluster-api"
+	mapiNamespace string = "openshift-machine-api"
+)
+
+var (
+	// errPlatformNotSupported is returned when the platform is not supported.
+	errPlatformNotSupported = errors.New(`error determining InfrastructureMachine
+	type, platform not supported`)
+)
+
 // MachineSyncReconciler reconciles CAPI and MAPI machines.
 type MachineSyncReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
+
+	platform      configv1.PlatformType
+	capiNamespace string
+	mapiNamespace string
 }
 
 // SetupWithManager sets the CoreClusterReconciler controller up with the given manager.
 func (r *MachineSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	infrastructureMachine, err := getInfrastructureMachineFromProvider(r.platform)
+	if err != nil {
+		return fmt.Errorf("failed to get InfrastructureMachine from Provider: %w", err)
+	}
+
+	if len(r.capiNamespace) == 0 {
+		r.capiNamespace = capiNamespace
+	}
+
+	if len(r.mapiNamespace) == 0 {
+		r.mapiNamespace = mapiNamespace
+	}
+
+	// I'm unsure of how to scope this to just the CAPI namespace
```

Comment:
Add a predicate to both `Watches` that restricts to events from the `r.capiNamespace`?

And the caches can be restricted to the correct namespaces too in the manager options in the main.go